torch/distributed/tensor/parallel/style.py

```diff
@@ -44,15 +45,36 @@ class PairwiseParallel(ParallelStyle):
     """
     PairwiseParallel concatenate colwise and rowwise styles as a fixed
     pair like what Megatron-LM(https://arxiv.org/abs/1909.08053) is doing.
-    We assume both input and output needs to a replicate DTensor.
+    We assume both input and output need to a replicate DTensor.
 
     .. warning::
         PairwiseParallel only supports ``nn.Multihead Attention``,
         ``nn.Transformer`` or even-number-layer MLP for now.
     """
 
+    def __init__(self, _prepare_input=None, _prepare_output=None) -> None:
+        _prepare_input = (
+            make_input_replicate_1d if _prepare_input is None else _prepare_input
+        )
+        _prepare_output = (
+            make_output_tensor if _prepare_output is None else _prepare_output
+        )
+        super().__init__(_prepare_input, _prepare_output)
+
+
+class PairwiseSequenceParallel(PairwiseParallel):
+    """
+    PairwiseSequenceParallel concatenate colwise and rowwise styles as a fixed
+    pair like what Megatron-LM(https://arxiv.org/pdf/2205.05198.pdf) is doing.
+    We assume both input and output need to a sharded DTensor.
+
+    .. warning::
+        PairwiseSequenceParallel only supports ``nn.Multihead Attention``,
+        ``nn.Transformer`` or even-number-layer MLP for now.
+    """
+
     def __init__(self) -> None:
-        super().__init__(make_input_replicate_1d, make_output_tensor)
+        super().__init__(make_input_reshard_replicate, make_output_shard_1d)
```

Comment:
For output, we are transforming it to Shard(0) DTensor, shouldn't we also transform it to a torch.Tensor like the PiarwiseParallel?

Reply:
OK, we then need a new output function. Let me add it.